internal/util/debug/debug.go

```diff
@@ -100,25 +116,84 @@ func Listen(opts *ListenOpts) (*Handler, error) {
 		}),
 	))
 
+	http.HandleFunc("/debug/archive", func(rw http.ResponseWriter, req *http.Request) {
+		rw.Header().Set("Content-Type", "application/zip")
+
+		rw.Header().Set(
+			"Content-Disposition",
+			fmt.Sprintf("attachment; filename=FerretDB-debug-%d.zip", time.Now().UnixMilli()),
+		)
+
+		ctx := req.Context()
+		zipWriter := zip.NewWriter(rw)
+
+		defer func() {
+			if err := zipWriter.Close(); err != nil {
+				l.ErrorContext(ctx, "Archive handler failed", logging.Error(err))
+				return
+			}
+		}()
+
+		metricsFile, err := zipWriter.Create("metrics")
+		if err != nil {
+			l.ErrorContext(ctx, "Archive handler failed", logging.Error(err))
+			return
```

Comment:
What does the response to the user look like in this case and all the other cases when an error is returned? An empty zip file?

As this is a debug handler, I don't think it's super important to return something particular, but on the other hand, it's nice if the response is clear (like `500` or something). However, I'm not sure if it is possible to return something like 500 page, as we have already set some content-type headers. So, this is not critical, just some thinking.

Reply:
The request will just end with StatusOK. I think we could use multipart here to handle error properly, but I'd suggest doing that in separate PR. @FerretDB/core WDYT?

Reply:
and about multipart? :)

Reply:
I don't know how one can send a file using the multipart encoding, but I know for sure that nothing would handle that response. Definitely not `curl` or browser that would be used to download that debug archive.